### ISSUE TITLE
300个openimage权重

### ISSUE THREAD

**Issue by LuZeking** [LuZeking/hello-world#19]
from dataSet import *
dataset = DogCat('data/dogcat/', transform=transform)

from torch.utils.data import DataLoader
# 狗的图片被取出的概率是猫的概率的两倍
# 两类图片被取出的概率与weights的绝对大小无关，只和比值有关
weights = [2 if label == 1 else 1 for data, label in dataset]

print(weights)

from torch.utils.data.sampler import  WeightedRandomSampler
sampler = WeightedRandomSampler(weights,\
                                num_samples=9,\
                                replacement=True)
dataloader = DataLoader(dataset,
                        batch_size=3,
                        sampler=sampler)
for datas, labels in dataloader:
    print(labels.tolist())
————————————————
版权声明：本文为CSDN博主「蓝鲸123」的原创文章，遵循 CC 4.0 BY-SA 版权协议，转载请附上原文出处链接及本声明。
原文链接：https://blog.csdn.net/TH_NUM/article/details/80877772

Screwdriver
Light switch
Doughnut
Toilet paper
Wrench
Toaster
Tennis ball
Radish
Pomegranate
Kite
Table tennis racket
Hamster
Barge
Shower
Printer
Snowmobile
Fire hydrant
Limousine
Whale
Microwave oven
Asparagus
Lion
Spatula
Torch
Volleyball
Ambulance
Chopsticks
Raccoon
Blue jay
Lynx
Dice
Filing cabinet
Ruler
Power plugs and sockets
Bell pepper
Binoculars
Pretzel
Hot dog
Missile
Common fig
Croissant
Adhesive tape
Slow cooker
Dog bed
Harpsichord
Billiard table
Alpaca
Harbor seal
Grape
Nail
Paper towel
Alarm clock
Guacamole
Starfish
Zebra
Segway
Sea turtle
Scissors
Rhinoceros
Kangaroo
Jaguar
Leopard
Dumbbell
Envelope
Winter melon
Teapot
Camel
Beaker
Brown bear
Toilet
Teddy bear
Briefcase
Stop sign
Tiger
Cabbage
Giraffe
Polar bear
Shark
Rabbit
Swim cap
Pressure cooker
Kitchen knife
Submarine sandwich
Flashlight
Penguin
Snake
Zucchini
Bat
Food processor
Ostrich
Sea lion
Goldfish
Elephant
Rocket
Mouse
Oyster
Digital clock
Otter
Dolphin
Punching bag
Corded phone
Tennis racket
Pancake
Mango
Crocodile
Waffle
Computer mouse
Kettle
Tart
Oven
Banana
Cheetah
Raven
Frying pan
Pear
Fox
Skateboard
Rugby ball
Watermelon
Flute
Canary
Door handle
Saxophone
Burrito
Suitcase
Roller skates
Dagger
Seat belt
Washing machine
Jet ski
Sombrero
Pig
Drinking straw
Peach
Tortoise
Towel
Tablet computer
Cucumber
Mule
Potato
Frog
Bear
Lighthouse
Belt
Baseball bat
Racket
Sword
Bagel
Goat
Lizard
Parrot
Owl
Turkey
Cello
Knife
Handgun
Carrot
Hamburger
Grapefruit
Tap
Tea
Bull
Turtle
Bust
Monkey
Wok
Broccoli
Pitcher
Whiteboard
Squirrel
Jug
Woodpecker
Pizza
Surfboard
Sofa bed
Sheep
Candle
Muffin
Cookie
Apple
Chest of drawers
Skull
Chicken
Loveseat
Baseball glove
Piano
Waste container
Barrel
Swan
Taxi
Lemon
Pumpkin
Sparrow
Orange
Tank
Sandwich
Coffee
Juice
Coin
Pen
Watch
Eagle
Goose
Falcon
Christmas tree
Sunflower
Vase
Football
Canoe
High heels
Spoon
Mug
Swimwear
Duck
Cat
Tomato
Cocktail
Clock
Cowboy hat
Miniskirt
Cattle
Strawberry
Bronze sculpture
Pillow
Squash
Traffic light
Saucer
Reptile
Cake
Plastic bag
Studio couch
Beer
Scarf
Coffee cup
Wine
Mushroom
Traffic sign
Camera
Rose
Couch
Handbag
Fedora
Sock
Computer keyboard
Mobile phone
Ball
Balloon
Horse
Boot
Fish
Backpack
Skirt
Van
Bread
Glove
Dog
Airplane
Motorcycle
Drink
Book
Train
Flower
Carnivore
Human ear
Toy
Box
Truck
Wheel
Aircraft
Bus
Human mouth
Sculpture
Shirt
Hat
Vehicle registration plate
Guitar
Sun hat
Bottle
Luggage and bags
Trousers
Bicycle wheel
Suit
Bowl
Man
Flowerpot
Laptop
Boy
Picture frame
Bird
Car
Shorts
Woman
Platter
Tie
Girl
Skyscraper
Person
Flag
Jeans
Dress
 

0.020735365
0.024667934
0.00211648
0.004967848
0.008777547
0.026995097
0.00851631
0.00339038
0.004030254
0.001727947
0.002634881
0.002783541
0.001946585
0.008567306
0.005745944
0.004443292
0.003366447
0.004415865
0.00174056
0.003207938
0.021354331
0.001271769
0.026495188
0.071537008
0.004030254
0.003342851
0.002587234
0.004183451
0.006220609
0.006220609
0.002767389
0.004208059
0.006415875
0.005745944
0.002518909
0.013247594
0.006358845
0.00348961
0.003455894
0.00711811
0.003649847
0.006473937
0.011922835
0.005545505
0.007451772
0.001681246
0.001900053
0.000813383
0.005524093
0.013890681
0.007651017
0.010838941
0.00671709
0.002649519
0.001941303
0.002679289
0.001611194
0.004348754
0.00220453
0.002088672
0.002725219
0.00186052
0.004950658
0.008618917
0.043355763
0.002554893
0.001339644
0.009055317
0.002778136
0.001405442
0.001067717
0.009732926
0.003996481
0.001445192
0.007153701
0.001259454
0.002999455
0.002968341
0.001440826
0.002968341
0.102195726
0.004362013
0.005399019
0.018110635
0.000629173
0.00132109
0.001693184
0.003506716
0.009230582
0.002805373
0.001021228
0.001098879
0.000606246
0.002360957
0.001912754
0.001987139
0.009667163
0.001984383
0.001209417
0.005109786
0.003447567
0.001605769
0.002644621
0.005020141
0.001640757
0.002416791
0.00242088
0.002412715
0.001679272
0.002349327
0.001530203
0.00220453
0.002931845
0.003941433
0.002867215
0.003012085
0.001056677
0.00619368
0.002654434
0.004721915
0.005221679
0.002412715
0.001231274
0.005610746
0.002441536
0.00051281
0.004322478
0.003289058
0.002488244
0.002805373
0.002384567
0.001322311
0.006878558
0.00335068
0.000907254
0.005502847
0.001633265
0.001757666
0.001490354
0.004147073
0.001132811
0.005816017
0.001039027
0.003706581
0.001452528
0.006473937
0.003414654
0.002762047
0.000765511
0.000757005
0.000757005
0.001405442
0.002278249
0.000765511
0.001777317
0.002591921
0.001646421
0.001166999
0.001687194
0.001291282
0.001481097
0.001207376
0.008724025
0.001446653
0.000540514
0.002012293
0.001923038
0.004585706
0.001488803
0.000770873
0.002821973
0.003532692
0.000831342
0.000767976
0.00099288
0.000697582
0.000478508
0.000404621
0.000569336
0.000689513
0.001006854
0.001222855
0.00063958
0.002330196
0.000979288
0.001181453
0.000853155
0.000902106
0.000474856
0.00044893
0.001105672
0.000391662
0.001052789
0.000377205
0.001162258
0.001363909
0.000695547
0.000586609
0.000655401
0.000991504
0.000829896
0.001070112
0.00022599
0.000981977
0.00044893
0.000323258
0.000661155
0.000494552
0.000398313
0.000567304
0.000932686
0.000687195
0.000193187
0.000144344
0.000125482
0.000324504
0.000340896
0.001464422
0.000559757
0.001669475
0.000190664
0.000261513
0.000544836
0.000446967
0.006503364
0.000605476
0.000602417
0.003327303
0.000279332
0.001705292
0.000786121
0.000231924
0.000663917
0.000305192
0.000253228
0.000527559
0.000343763
0.000280868
0.000244279
0.000351361
0.000623688
0.000450201
0.001170818
0.000356171
0.000282086
0.000454203
0.000203057
0.000158425
0.000523697
0.000118537
0.001322311
0.001356152
0.000208077
0.000463023
0.001399941
6.52293E-05
8.58685E-05
0.000119707
6.90012E-05
0.000118961
0.000145327
1.59354E-05
0.000651521
0.000295791
3.23975E-05
0.000325094
0.000133154
9.20587E-06
0.001228103
0.00016092
0.000141981
5.3848E-05
0.000199601
0.000133527
0.000463773
5.89656E-05
0.000249171
5.80728E-05
0.000749079
0.000196288
3.25709E-05
1.4249E-05
0.000405079
1.25783E-05
9.56058E-05
0.000169338
1.78159E-05
0.000150478
6.04045E-05
8.97381E-06
0.000113434
1.22041E-05
0.000476596
0.000159806
8.23272E-06
2.34256E-05
9.5916E-06
6.8958E-05
2.34256E-05
3.00544E-05
1.000000099
